src/app/modules/alert/alert.component.ts

```diff
@@ -11,7 +11,22 @@ export class AlertComponent {
     @Input() hideIcon = false;
     @Input() hideClose = false;
     @Input() dataAutoId: string;
+    @Input() dataTest: string;
 
     hide: boolean;
+    dataTestVals = {
+      outer: null,
+      close: null,
+      heading: null,
+    };
 
+    ngOnInit() {
+      if (this.dataTest) {
+        this.dataTestVals = {
+          outer: `${this.dataTest}-${this.variation}`,
+          close: `${this.dataTest}-close`,
+          heading: `${this.dataTest}-heading`,
+        };
+      }
+    }
```

Comment:
This seems like a lot overhead for developers when the testers can just as easily get the right selector without multiple test ids.  If we went the route of directives...I think a lot of this can go away.  So say we have this alert component, the test Id could simply be on the component itself:

```
<app-alert testId='profile-page-alert'>
...
<app-alert>
```

The testers could then do something like (this may not be the exact syntax):

```
 cy.get(app-alert[testId='profile-page-alert'])
  .find('button') 
  .click()
```

This isn't brittle because we're already looking in the context of app-alert and there's only one button so it just seems like overkill to add another test id.
Say it was a more complex component with multiple buttons, we could then select say the close button:
`.find('button.ds-c-dialog__close')`. In the case of the header, they could use `.find('.ds-c-alert__heading')`.

Again I don't think these are brittle selectors because I doubt the design system is ever going to change these classes and they're not coupled to a style.

So basically, I'm wondering if we can use a happy medium where we use non-brittle (class) selectors where they make sense but use the test ids to make it easier to get to the actual component that's being tested.

If none of this makes sense, let me know and we can huddle and talk through it.

Reply:
Another thought since our chat, and while cleaning up to go this direction ... suggesting usage of inner `class` or other existing attributes to find internal elements of `shared UI` or even `app` components will encourage better practices anyhow.

We'll probably want to provide this angle as motivation to create specific "wrapper" CSS classes that describe their purpose. We do that better in this repo than we do in other apps, and it would be a good habit to spread, not just for test automation.